File: cpp/examples/linux/accelerometer/ADXL345/minimal/main.cpp
#include <cstdio>
#include <cstdlib>
#include <cmath>
#include <unistd.h>
#include "I2CConnectionLinux.h"
#include "ADXL345.h"

int main() {
    const char* bus_env  = getenv("I2C_BUS");
    const char* addr_env = getenv("I2C_ADDR");
    int     bus  = bus_env  ? atoi(bus_env)       : 1;
    uint8_t addr = addr_env ? (uint8_t)strtol(addr_env, nullptr, 0) : 0x53;
    I2CConnectionLinux connection(bus, addr);

    ADXL345Minimal accel(connection);                       // Create ADXL345 driver, (connection, spi=false)

    while (true) {
        float x, y, z;
        accel.read(x, y, z);                                // Read 3-axis acceleration, (x, y, z) → g, g, g
        printf("x=%.3f y=%.3f z=%.3f g\n", (double)x, (double)y, (double)z);
        usleep(100000);
    }
    return 0;
}

File: rust/examples/linux/accelerometer/adxl345/minimal/src/main.rs
use linux_embedded_hal::I2cdev;
use periph::chips::accelerometer::Adxl345Minimal;

fn main() {
    let i2c_bus: u8 = std::env::var("I2C_BUS").ok().and_then(|v| v.parse().ok()).unwrap_or(1);
    let addr: u8 = std::env::var("I2C_ADDR")
        .ok()
        .and_then(|v| u8::from_str_radix(v.trim_start_matches("0x"), 16).ok())
        .unwrap_or(0x53);

    let dev  = I2cdev::new(format!("/dev/i2c-{}", i2c_bus)).expect("open i2c bus");
    let mut accel = Adxl345Minimal::new(dev, addr, false).expect("init ADXL345"); // Create ADXL345 driver, (i2c, addr=0x53, spi=false)

    for _ in 0..10 {
        let (x, y, z) = accel.read().expect("read acceleration");   // Read 3-axis acceleration, () → (f32, f32, f32) g
        println!("x={:.3} y={:.3} z={:.3} g", x, y, z);
        std::thread::sleep(std::time::Duration::from_millis(100));
    }
}